File: packages/shared/src/player.ts
export interface PlayerStats {
  matches: number;
  acs: number;
  kd: number;
  hs_percent: number;
  adr: number;
  win_rate: number;
}

export interface PlayerProfile {
  id: string;
  display_name: string;
  riot_id: string;
  team: string | null;
  region: string;
  rank: string;
  stats: PlayerStats;
}

export interface PlayerComparison {
  players: PlayerProfile[];
  notes: string;
}

export interface HealthResponse {
  status: "ok" | "degraded";
  service: string;
  database: "connected" | "disconnected";
}

File: apps/api/app/schemas/player.py
from pydantic import BaseModel


class PlayerStats(BaseModel):
    matches: int
    acs: float
    kd: float
    hs_percent: float
    adr: float
    win_rate: float


class PlayerProfile(BaseModel):
    id: str
    display_name: str
    riot_id: str
    team: str | None
    region: str
    rank: str
    stats: PlayerStats


class PlayerComparison(BaseModel):
    players: list[PlayerProfile]
    notes: str
